service/controller/v14/key/key.go

```diff
@@ -227,6 +229,14 @@ func EC2ServiceDomain(customObject v1alpha1.AWSConfig) string {
 	return domain
 }
 
+func BaseDomain(customObject v1alpha1.AWSConfig) string {
+	// TODO remove other zones and make it a BaseDomain in the CR.
+	// CloudFormation creates a separate HostedZone with the same name.
+	// Probably the easiest way for now is to just allow single domain for
+	// everything which we do now.
```

Comment:
Is this TODO tracked somewhere?

Reply:
https://github.com/giantswarm/giantswarm/issues/2383#issuecomment-372673728

I also discussed that with Puja. This is important to me to get DNS straight. I actually made effort in a few places already towards that goal. One example is https://github.com/giantswarm/kubernetesd/pull/385

Reply:
I think it would be good if we could establish a best practise to create issues or reference the TODOs in some place and link to this place in the TODO comment in the code. Then I do not have to ask all the time and we are safe.

Reply:
I agree it would be good and I would gladly follow. But I don't understand why my PR should be blocked when we don't have one yet.

Reply:
Lets just make it a thing now. Or where do you want to have this written down so we can apply it? Also no blocking. So far we are talking here.

Reply:
With this TODO this is tricky to say when we can do it. This is more of informational thing.
In scope of https://github.com/giantswarm/giantswarm/issues/2383 `.Spec.AWS.HostedZones.API.Name` will be removed so then it will resolve. No need to track it somewhere.